cypress/pageobject/Login/LoginPage.ts

```diff
@@ -88,6 +111,68 @@ class LoginPage {
   verifyForgotPasswordHeading(text: string[]): void {
     cy.verifyContentPresence("#forgot-password-heading", text);
   }
+
+  clickContributeOnGitHub(): void {
+    cy.get(this.contributeOnGitHubLink).scrollIntoView().click();
+  }
+
+  clickThirdPartyLicense(): void {
+    cy.get(this.thirdPartyLicenseLink).scrollIntoView().click();
+  }
+
+  switchLanguage(languageCode: string): void {
+    cy.get(this.languageSelector).select(languageCode);
+  }
+
+  getSubmitButton() {
+    return cy.get(this.submitButtonSelector);
+  }
+
+  getSidebarItems() {
+    return cy.get(`${this.sidebarSelectors.care}, ${this.sidebarSelectors.goal}, ${this.sidebarSelectors.footer_body}`);
+  }
+
+  switchLanguageAndVerifyButtonText(languageMappings: LanguageMapping): void {
+    Object.entries(languageMappings).forEach(([languageCode, texts]) => {
+      cy.get(this.languageSelector)
+        .find(`option[value="${languageCode}"]`)
+        .should("exist")
+        .then(($option) => {
+          if ($option.length === 0) {
+            throw new Error(`Language option ${languageCode} not found`);
+          }
+          this.switchLanguage(languageCode);
+          cy.get(this.submitButtonSelector, { timeout: 10000 })
+            .should("be.visible")
+            .should("have.text", texts.login)
+            .should("not.be.disabled");
+        });
+    });
+  }
+
+  private verifySidebarElement(selector: string, expectedText: string): void {
+    cy.get(selector, { timeout: 10000 })
+      .should("be.visible")
+      .should("have.text", expectedText)
+      .should("not.be.disabled");
+  }
+
+  switchLanguageAndVerifySidebars(languageMappings: SidebarMapping): void {
+    Object.entries(languageMappings).forEach(([languageCode, texts]) => {
+      cy.get(this.languageSelector)
```

Comment:
language selector doesn't exist, from this code I assume you are referring to language selector from the profile page, but this test is not being called for a logged in user.

Additionally profile page was just replaced anyways (but you need to add the selector regardless). Either that or rewrite the logic to use the language selection options on the login page.